src/ansys/mapdl/core/component.py

```diff
@@ -0,0 +1,157 @@
+from typing import Any
+import warnings
+import weakref
+
+import numpy as np
+
+from ansys.mapdl.core.mapdl import _MapdlCore
+
+ENTITIES_MAPPING = {
+    "NODE": "NSEL",
+    "NODES": "NSEL",
+    "ELEM": "ESEL",
+    "ELEMS": "ESEL",
+    "ELEMENTS": "ESEL",
+    "VOLU": "VSEL",
+    "AREA": "ASEL",
+    "LINE": "LSEL",
+    "KP": "KSEL",
+}
+
+VALID_ENTITIES = list(ENTITIES_MAPPING.keys())
+
+
+def _check_valid_pyobj_to_entities(items):
+    """Check whether the python objects can be converted to entities.
+    At the moment, only list and numpy arrays of ints are allowed.
+    """
+    if not isinstance(items, (list, np.ndarray)):
+        raise TypeError(
+            "Only list or numpy arrays are allowed for component definitions."
+        )
+
+    if not all([isinstance(each, (int, np.int_)) for each in items]):
+        raise ValueError("Only integers are allowed for component definition.")
+
+
+class Component:
+    def __setitem__(self, __key: Any, __value: Any) -> None:
+        if __key not in ["type", "items"]:
+            raise KeyError(f"The key '{__key}' is not allowed in 'Component' class.")
+
+        if __key == "type":
+            if not isinstance(__value, str) or __value.upper() not in VALID_ENTITIES:
+                raise ValueError(
+                    f"The value '{__value[0]}' is not allowed for 'type' definition."
+                )
+        return super().__setitem__(__key, __value)
+
+    @property
+    def type(self):
+        return super().__getitem__("type")
+
+    @property
+    def items(self):
+        return super().__getitem__("items")
+
+
+class ComponentManager(dict):
+    def __init__(self, mapdl):
+        if not isinstance(mapdl, _MapdlCore):
+            raise TypeError("Must be implemented from MAPDL class")
+        self._mapdl_weakref = weakref.ref(mapdl)
+        self._comp = None
+
+    @property
+    def _mapdl(self):
+        """Return the weakly referenced instance of mapdl"""
+        return self._mapdl_weakref()
+
+    def _set_log_level(self, level):
+        self._mapdl.set_log_level(level)
+
+    @property
+    def _log(self):
+        return self._mapdl._log
+
+    def __getitem__(self, __key: Any) -> Any:
+        self._comp = self._mapdl._parse_cmlist()
+        try:
+            cmtype = self._comp[__key.upper()]
+        except KeyError:
+            raise KeyError(
+                f"The component named '{__key}' does not exist in the MAPDL instance."
+            )
+        return self._mapdl._parse_cmlist_indiv(__key, cmtype)
+
+    def __setitem__(self, __key: Any, __value: Any) -> None:
```

Comment:
I'd avoid using leading underscores, there's really no need for this in a dunder method.

Reply:
For some reason, they were suggested by VSCode autocompletation and I kept it like that. I will correct that.